### PR TITLE
Fixed a null reference exception in Map.Save

### PR DESCRIPTION
**Pull request by Mailaender**
Discovered via Coverity.

### CODE REVIEW

**Comment by pchote** [comment]
"If we don't have an existing container in `Save()` then don't actually save the file" doesn't seem like a proper fix to me.

File: OpenRA.Game/Map/Map.cs [hunk 1]
```diff
@@ -493,7 +493,8 @@ public void Save(string toPath)
 			}
 
 			// Update existing package
-			Container.Write(entries);
+			if (Container != null)
```
Comment (pchote) [inline OpenRA.Game/Map/Map.cs]: A better fix would probably to add an `|| Container == null` case to the conditional above.

**Comment by Mailaender** [comment]
Updated.

File: OpenRA.Game/Map/Map.cs [hunk 2]
```diff
@@ -483,10 +483,11 @@ public void Save(string toPath)
 				}
 			}
 
-			// Saving the map to a new location
-			if (toPath != Path)
+			if (toPath != Path || Container == null)
 			{
-				Path = toPath;
+				// Saving the map to a new location
+				if (Path != toPath)
```
Comment (pchote) [inline OpenRA.Game/Map/Map.cs]: This check is unnecessary.  If `Path == toPath` then reassigning the variable doesn't change anything.

Comment (Mailaender) [inline OpenRA.Game/Map/Map.cs]: Removed the superfluous check.

**Comment by pchote** [comment]
:+1:

**Comment by obrakmann** [comment]
:+1:

**Comment by obrakmann** [comment]
[Changelog](https://github.com/OpenRA/OpenRA/wiki/Changelog-%28bleed%29/_compare/57db26c9f25667b8afcef570f657e30cd5490e24%5E...57db26c9f25667b8afcef570f657e30cd5490e24)

**Pull request merged**